okio/jvm/src/main/java/okio/Okio.kt

```diff
@@ -136,8 +143,124 @@ private class BlackholeSink : Sink {
 @Throws(IOException::class)
 fun Socket.sink(): Sink {
   val timeout = SocketAsyncTimeout(this)
-  val sink = OutputStreamSink(getOutputStream(), timeout)
-  return timeout.sink(sink)
+  val sink = SelectableSocketSink(this, timeout)
+  return sink // TODO(jwilson): restore timeouts.
+}
+
+private class SelectableSocketSource(
+  socket: Socket,
+  val timeout: SocketAsyncTimeout
+) : Source {
+  val channel: SocketChannel = socket.channel!!
+
+  override fun read(sink: Buffer, byteCount: Long): Long {
+    return runBlocking {
```

Comment:
I'd originally used that and it broke because the channel has a blocking mode and you need to change it. I might just change it to use the blocking mode.